### DISCUSSION TITLE
[OP Stack Development] What can we modify at most to OP Stack to still be eligible for joining the Superchain?

### DISCUSSION THREAD

**Discussion by upnodedev** [ethereum-optimism/developers#233] [General]
### Discussion Topic

We're curious about what can we change most to OP Stack to still be eligible for joining the Superchain?
* Can we use Celestia as DA instead of ETH mainnet?
* Can we use an espresso-shared sequencer?
* Can we modify the gas formula?
* Can we add more transactions to the sequencer in each block in addition to L1Block tx?
* Can we become L3 instead of L2 like settled on Optimism or Base instead of ETH?
* Can we add predeploy
* Can we add precompile
* etc

**Comment by hash1go** [comment]
For a chain to join the Superchain, it needs to comply with the Law of Chains. 
https://gov.optimism.io/t/final-law-of-chains-v0-1/6514

Additionally, the Version note of the Law of Chains includes the following wording, which I think will be useful.
> The Law of Chains is a living document, and should evolve alongside protocol innovation across the Collective. Version 0.1 of the Law of Chains is immediately applicable to the current, single-sequencer release of the OP Stack, and establishes fundamental principles: core expectations of users and economic participants, homogenous blockspace, and universal upgrades on a shared standard. Anticipated improvements to the OP Stack that remain technologically unspecified (e.g., shared sequencing and zero-knowledge proofs) are correspondingly economically unspecified in this initial version. As the Law of Chains evolves, two critical freedoms should be preserved: (1) the freedom of Optimism Governance to introduce new protocol constructions with different technological and economic specifications, and (2) the freedom of ecosystem participants to choose, at the appropriate time, with full information, and without coercion, whether to opt into those constructions.

**Comment by sbvegan** [comment]
Hey @upnodedev! I raised these questions internally and our product team has heard your feedback. They're working on a crisp spec for Superchain compatibility. When thats ready, I'll make sure its relayed here :)
